client/src/constants/naviagationUrls.js

```diff
@@ -1,4 +1,5 @@
-export const SEARCH_URL = "/search";
+export const SEARCH_URL = "/search/:category?";
```

Comment:
What does the question mark here do?

Reply:
this means this param `category` is optional for this route and react-router will handle both cases(with or without will render the related component)

Reply:
thanks!